packages/bridge-ui-v2/src/libs/token/isDeployedCrossChain.test.ts

```diff
@@ -0,0 +1,63 @@
+import { getContract, type GetContractResult } from '@wagmi/core';
+import { zeroAddress } from 'viem';
+
+import { isDeployedCrossChain } from './isDeployedCrossChain';
+import { type Token, TokenType } from './types';
+
+vi.mock('@wagmi/core');
+vi.mock('$abi');
+
+let mockToken: Token;
+
+const mockTokenVaultContract = {
+  read: {
+    canonicalToBridged: vi.fn(),
+    isBridgedToken: vi.fn(),
+    bridgedToCanonical: vi.fn(),
+  },
+} as unknown as GetContractResult<readonly unknown[], unknown>;
+
+describe('isDeployedCrossChain', () => {
+  const srcChainId = 1;
+  const destChainId = 2;
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+    vi.mocked(getContract).mockReturnValue(mockTokenVaultContract);
+    mockToken = {
+      name: 'MockToken',
+      addresses: {
+        1: '0x123456',
+        2: '0x654321',
+      },
+      symbol: 'MOCK',
+      decimals: 18,
+      type: TokenType.ERC20,
+    };
+  });
+
+  it('should return true if configured', async () => {
+    // When
+    const result = await isDeployedCrossChain({
+      token: mockToken,
+      srcChainId,
+      destChainId,
+    });
+
+    // Then
+    expect(result).toEqual(true);
+  });
+
+  it('should return false if not configured', async () => {
+    mockToken.addresses[destChainId] = zeroAddress;
```

Comment:
The reason this test works is because `getCrossChainAddress() `reads the configured destinationAddress (`zeroAddress` in this case) and treats it as valid. This is actually an error and should be fixed.

[this line](https://github.com/taikoxyz/taiko-mono/blob/95337bce36c5e9e4b07955501baadec8ddf93a2c/packages/bridge-ui-v2/src/libs/token/getCrossChainAddress.ts#L35) should also check if it is `!zeroAddress`.

You are only testing preconfigured tokens that have some form of address at the destination. What we actually need it for, is when we don't have that address. We check if we can find one via `getCrossChainAddress(),` if we do, it is deployed. So the test should also handle tokens without any destination address set.

Reply:
@xiaodino

Reply:
@KorbinianK  Line 31 and 32 have checked  if it is ```!zeroAddress```.